src/ActionModal/components/ModalWithdrawRewards.vue

```diff
@@ -146,6 +137,27 @@ export default {
   methods: {
     open() {
       this.$refs.actionModal.open()
+    },
+    getTop5ValidatorsRewards(top5validators) {
+      return this.rewards
+        .filter(({ validator }) =>
+          top5validators.includes(validator.operatorAddress)
+        )
+        .reduce((totalRewardsAgreggator, { amount, denom }) => {
+          const rewardDenom = denom
+          const sameDenomReward = totalRewardsAgreggator.find(
```

Comment:
better use a dictionary?

Reply:
You mean instead of the reduce?

Reply:
in the reduce: look at `totalRewardsPerDenom` in `TmBalance`